File: java/ankit_kanyal/devnest_problems/phase_2/SumOfLeftLeaves.java
// link- https://devsnest.in/algo-challenges/sum-of-left-leaves?tab=question

static int solve(TreeNode root){
//CODE HERE 

int ans=0;
        
    if(root==null){
        return 0;
    }
    if(root.left!=null && root.left.left==null && root.left.right==null){
        ans+=root.left.val;
    }
    ans+=solve(root.left);
    ans+=solve(root.right);
        
    return ans;
}

File: Python/vedant_bhatnagar/DevsnestProblems_THA/Phase_2/Trees/sumofleftleaves.py
def solve(root):
    # CODE HERE
    q = deque([None, root])
    if not root or (not root.left and not root.right):
        return 0
    sum = 0
    while q:
        n = q.pop()
        if n:
            if n.left:
                q.appendleft(n.left)
            if n.right:
                q.appendleft(n.right)
            if n.left and n.left.left == None and n.left.right == None:
                sum += n.left.val
        else:
            if q:
                q.appendleft(None)
            else:
                break
    return sum